sdk/mixedreality/azure-mixedreality-authentication/azure/mixedreality/authentication/_shared/aio/mixedreality_account_key_credential.py

```diff
@@ -0,0 +1,54 @@
+# --------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# --------------------------------------------------------------------------
+from typing import TYPE_CHECKING
+
+from datetime import date, datetime
+
+from azure.core.credentials import AccessToken
+
+if TYPE_CHECKING:
+    # pylint: disable=unused-import,ungrouped-imports
+    from typing import Any
+    from azure.core.credentials import AzureKeyCredential
+
+ACCOUNT_KEY_VALID_YEARS = 10
+
+class MixedRealityAccountKeyCredential(object):
+    """ Represents an object used for Mixed Reality account key authentication.
+
+    :param str account_id: The Mixed Reality service account identifier.
+    :param AzureKeyCredential account_key: The Mixed Reality service account primary or secondary key credential.
+    """
+
+    def __init__(self, account_id, account_key):
+        # type: (str, AzureKeyCredential) -> None
+        self.account_id = account_id
+        self.account_key = account_key
+
+    async def get_token(self, *scopes: str, **kwargs: "Any") -> "AccessToken": #pylint: disable=unused-argument
+        token = self.account_id + ":" + self.account_key.key
+
+        # No way to know when an access token might expire, so we'll set it to be
+        # 10 years in the future.
+        expiration_date = _add_years(datetime.now(), ACCOUNT_KEY_VALID_YEARS)
```

Comment:
Ah. The comment is a bit confusing. I'll update it. Essentially, an account key could be good forever. It doesn't expire automatically. An account key can be "rolled" which invalidates it, but there's no way to know when that might happen, so we set the AccessToken we're creating here to expire in 10 years. If someone manages to hold on to this particular object in memory for 10 years, i'll be amazed, but it could technically work for up to 10 years.